contracts/Content/UniqueContent.sol

```diff
@@ -0,0 +1,173 @@
+        _burn(_uniqueId);
+
+        uniqueAssetInfo[_uniqueId].creator = address(0);
+        uniqueAssetInfo[_uniqueId].contentAddress = address(0);
+        uniqueAssetInfo[_uniqueId].tokenId = 0;
+        uniqueAssetInfo[_uniqueId].version = 0;
+        uniqueAssetInfo[_uniqueId].creatorLocked = false;
+        delete uniqueAssetInfo[_uniqueId].uniqueAssetUri;
+        delete tokenRoyalties[_uniqueId];
+        
+        emit Burn(_msgSender(), _uniqueId);
+    }
+
+    /**
+    * @dev Returns the uri of a specific version of the original asset the unique asset is based on
+    * @param _uniqueId uint256 ID of token to query original asset uri of
+    * @param _version version number of token to query
+    */
+    function originalAssetUri(uint256 _uniqueId, uint256 _version) external view override returns (string memory) {
+        require(_exists(_uniqueId), "Unique Id does not exist");
+        uint256 _tokenId = uniqueAssetInfo[_uniqueId].tokenId;
+        return IContent(uniqueAssetInfo[_uniqueId].contentAddress).uri(_tokenId, _version);
+    }
+
+    /**
+    * @dev Returns the unique asset uri of a specific version
+    * @param _uniqueId uint256 ID of token to query
+    * @param _version version number of token to query
+    */
+    function uri(uint256 _uniqueId, uint256 _version) external view override returns (string memory) {
+        require(_exists(_uniqueId), "Unique Id does not exist");
+        if (_version > uniqueAssetInfo[_uniqueId].version) {
+            _version = uniqueAssetInfo[_uniqueId].version;
+        } 
+        return uniqueAssetInfo[_uniqueId].uniqueAssetUri[_version];
+    }
+
+    /**
+    * @dev Adds a new version of the unique asset
+    * @param _uniqueId uint256 ID of the token to set its uri
+    * @param _uri string URI to assign
+    */
+    function setUniqueUri(uint256 _uniqueId, string memory _uri) external override {
+        require(_exists(_uniqueId), "Unique Id does not exist");
+        require(uniqueAssetInfo[_uniqueId].creator == _msgSender(), "You are not the creator of this token");
+        uniqueAssetInfo[_uniqueId].uniqueAssetUri.push(_uri);
+        uint256 _version = ++uniqueAssetInfo[_uniqueId].version;
+
+        emit UniqueUriUpdated(_uniqueId, _version);
+    }
+
+    /**
+    * @dev Returns the original asset's receiver address and royalty amount for a token sold at a certain sales price
+    * @param _uniqueId uint256 ID of token to query
+    * @param _salePrice price the asset is to be purchased for
+    */
+    function royaltyInfo(uint256 _uniqueId, uint256 _salePrice) external view override returns (address receiver, uint256 royaltyAmount){
+        require(_exists(_uniqueId), "Unique Id does not exist");
+        // grabs the original asset's content contract address and tokenId
+        address originalAddress = uniqueAssetInfo[_uniqueId].contentAddress;
```

Comment:
same for originalAddress and tokenId, don't bother with local variables. Just plug it in directly in the royaltyInfo call in line 115